### ISSUE TITLE
Add a SDL_Renderer graphics backend

### ISSUE THREAD

**Issue by nmlgc** [nmlgc/ssg#4]
Might as well go fully cross-platform if we decide to change anything. We're probably looking at 5+ ReC98 pushes there, though.

Depends on #5.

**Comment by nmlgc** [comment]
Received €225 in funding from [T0235](https://github.com/nmlgc/rec98.nmlgc.net/commit/a7d6a68cf8d5ae54ee61571ed2191bc57870dc67), along with any necessary dependencies.

**Comment by nmlgc** [comment]
Turns out that SDL 2 comes with [its own abstraction for 2D rendering](https://wiki.libsdl.org/SDL2/SDL_Renderer) that just happens to exactly cover everything we need for Shuusou Gyoku. This even includes all the game's Direct3D code, which restricts itself to alpha-blended, untextured, and pre-transformed vertex-colored triangles and lines. It's the exact abstraction I thought I had to write myself, and such a perfect match for this game that it would be foolish to directly write OpenGL. Especially since SDL can also target other graphics APIs that might be preferred on certain operating systems.
